src/Interpolators.f90

```diff
@@ -0,0 +1,162 @@
+  !> @copydetails Interpolators::Interp_3D
+  MODULE PROCEDURE Interp_3D
+ENDINTERFACE Interp
+!
+!===============================================================================
+CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief This routine takes a 1D data "table" and linearly interpolates to find
+!>        a desired point.
+!> @param labels axis labels at which data points in table are defined
+!> @param table contains data to be interpolated between defined at points
+!>        given in labels
+!> @param axis_p point at which interpolation needs to be done to return a value
+!> @param data_p interpolated data found through linear interpolation to the point axis_p
+FUNCTION Interp_1D(labels,table,axis_p) RESULT(data_p)
+  REAL(SRK),INTENT(IN) :: labels(:),table(:),axis_p
+  REAL(SRK) :: data_p,f(2)
+  INTEGER(SIK) :: i,N_i,i_p
+
+  CALL Get_points_and_weights(labels,axis_p,f,i_p,N_i)
+
+  !Get interpolated value
+  data_p=0.0_SRK
+  DO i=MAX(i_p-1,1),MIN(i_p,N_i)
+    data_p=data_p+f(i-i_p+2)*table(i)
+  ENDDO
+
+ENDFUNCTION Interp_1D
+!
+!-------------------------------------------------------------------------------
+!> @brief This routine takes a 2D data table and linearly interpolates to find
+!>        a desired point.
+!> @param labels1 axis labels at which data points in table are defined for 1st dimension
+!> @param labels2 axis labels at which data points in table are defined for 2nd dimension
+!> @param table contains data to be interpolated between defined at points
+!>        given in labels1 and labels2
+!> @param axis_p point at which interpolation needs to be done to return a value
+!> @param data_p interpolated data found through linear interpolation to the point axis_p
+FUNCTION Interp_2D(labels1,labels2,table,axis_p) RESULT(data_p)
+  REAL(SRK),INTENT(IN) :: labels1(:),labels2(:),table(:,:),axis_p(:)
+  REAL(SRK) :: data_p,f1(2),f2(2)
+  INTEGER(SIK) :: i,j,N_i,N_j,i_p,j_p
+
+  CALL Get_points_and_weights(labels1,axis_p(1),f1,i_p,N_i)
+  CALL Get_points_and_weights(labels2,axis_p(2),f2,j_p,N_j)
+
+  !Get interpolated value
+  data_p=0.0_SRK
+  DO j=MAX(j_p-1,1),MIN(j_p,N_j)
+    DO i=MAX(i_p-1,1),MIN(i_p,N_i)
+      data_p=data_p+f1(i-i_p+2)*f2(j-j_p+2)*table(i,j)
+    ENDDO
+  ENDDO
+
+ENDFUNCTION Interp_2D
+!
+!-------------------------------------------------------------------------------
+!> @brief This routine takes a 2D data table and linearly interpolates to find
+!>        a desired point.
+!> @param labels1 axis labels at which data points in table are defined for 1st dimension
+!> @param labels2 axis labels at which data points in table are defined for 2nd dimension
+!> @param labels3 axis labels at which data points in table are defined for 3rd dimension
+!> @param table contains data to be interpolated between defined at points
+!>        given in labels1, labels2, and labels 3
+!> @param axis_p point at which interpolation needs to be done to return a value
```

Comment:
I am really confused by this comment.  What does this do?

Reply:
ok... I get what it is now, but its still confusing  This is the wording in numpy.interp "The x-coordinates at which to evaluate the interpolated values."... I'd drop "x-" and its at least clearer.  I'm probably most confused by the word axis since it doesn't have anything to do with the axis (I'm thinking which direction in the array)